Core/Utilities/Glue2.py

```diff
@@ -39,188 +41,243 @@ def getGlue2CEInfo(vo, host):
+    shareInfoLists.setdefault(siteName, []).append(shareInfo['attr'])
+
+  siteInfo = __getGlue2ShareInfo(host, shareInfoLists)
+  if not siteInfo['OK']:
+    sLog.error("Could not get CE info for", shareID + "  " + siteInfo['Message'])
+    return siteInfo
+  siteDict = siteInfo['Value']
+  sLog.debug("Found Sites:\n%s" % pformat(siteDict))
+  sitesWithoutShares = set(siteDict) - listOfSitesWithPolicies
+  if sitesWithoutShares:
+    sLog.error("Found some sites without any shares", pformat(sitesWithoutShares))
+  else:
+    sLog.notice("Found information for all known sites")
   return S_OK(siteDict)
 
 
-def __getGlue2ShareInfo(host, shareEndpoints, shareInfoDict, cesDict):
+def __getGlue2ShareInfo(host, shareInfoLists):
   """ get information from endpoints, which are the CE at a Site
 
   :param str host: BDII host to query
-  :param list shareEndpoints: list of endpoint names
   :param dict shareInfoDict: dictionary of GLUE2 parameters belonging to the ComputingShare
-  :param dict cesDict: dictionary with the CE information,
-                       will be modified in this function to add the information of the share
   :returns: result structure S_OK/S_ERROR
   """
+  executionEnvironments = []
+  for _siteName, shareInfoDicts in shareInfoLists.items():
+    for shareInfoDict in shareInfoDicts:
+      executionEnvironment = shareInfoDict['GLUE2ComputingShareExecutionEnvironmentForeignKey']
+      if isinstance(executionEnvironment, basestring):
+        executionEnvironment = [executionEnvironment]
+      executionEnvironments.extend(executionEnvironment)
+  resExeInfo = __getGlue2ExecutionEnvironmentInfo(host, executionEnvironments)
+  if not resExeInfo['OK']:
+    sLog.error("Cannot get execution environment info for:",
+               str(executionEnvironments)[:100] + "  " + resExeInfo['Message'])
+    return resExeInfo
+  exeInfos = resExeInfo['Value']
+
+  siteDict = {}
+  for siteName, shareInfoDicts in shareInfoLists.items():
+    siteDict[siteName] = {'CEs': {}}
+    for shareInfoDict in shareInfoDicts:
+      ceInfo = {}
+      ceInfo['MaxWaitingJobs'] = shareInfoDict.get('GLUE2ComputingShareMaxWaitingJobs', '-1')  # This is not used
+      ceInfo['Queues'] = {}
+      queueInfo = {}
+      queueInfo['GlueCEStateStatus'] = shareInfoDict['GLUE2ComputingShareServingState']
+      queueInfo['GlueCEPolicyMaxCPUTime'] = str(int(shareInfoDict.get('GLUE2ComputingShareMaxCPUTime', 86400)) / 60)
+      queueInfo['GlueCEPolicyMaxWallClockTime'] = str(int(shareInfoDict
+                                                          .get('GLUE2ComputingShareMaxWallTime', 86400)) / 60)
+      queueInfo['GlueCEInfoTotalCPUs'] = shareInfoDict.get('GLUE2ComputingShareMaxRunningJobs', '10000')
+      queueInfo['GlueCECapability'] = ['CPUScalingReferenceSI00=2552']
+
+      try:
+        maxNOPfromCS = gConfig.getValue('/Resources/Computing/CEDefaults/MaxNumberOfProcessors', 8)
```

Comment:
This is really a limit specific to Glue2-discovered resources, done with the only purpose of limiting the mad values that are found there. So I would make it clear with a specific name, e.g. something like `GLUE2ComputingShareMaxSlotsPerJob_limit`

Reply:
Renamed the option